libgcrypt Error During cmake ..;

# What I did

1. `git clone https://github.com/greenbone/openvas-scanner`
2. `cd openvas-scanner`
3. `mkdir build`
4. `cd build`
5. `cmake ..`

<!--
If reporting an issue please try to provide the information asked below.

Before reporting an issue please:

1. Be aware that this is not a support forum. If your issue is rather a question
   than a bug report, please use our community forum at
   https://community.greenbone.net/c/gse instead.
7. Make sure that you're using the latest published GVM components for the
   release you are using: https://community.greenbone.net/t/about-the-source-edition-gse-category/176
8. Check the list of issues whether it isn't already reported.
9. Read 1. again and if you still believe you found a software bug please
   continue to file this issue. If you are in doubt use
   https://community.greenbone.net/c/gse instead.

Thanks for your help to keep the communication channels clean and consistent!
-->

### Expected behavior

`cmake` creates `Makefile` and necessary build infrastructure.
<!--
  How did you expect openvas to behave?
  Please write down how the web interface should work in your opinion.
-->

### Actual behavior

I got the following error:

```
...
-- Checking for module 'gnutls>=3.2.15'
--   Found gnutls, version 3.8.6
-- Checking for module 'libopenvas_wmiclient>=1.0.5'
--   Package 'libopenvas_wmiclient', required by 'virtual:world', not found
-- Looking for pcap...
-- Looking for netsnmp...
-- Looking for netsnmp... /usr/lib/libnetsnmp.so
-- Looking for libgcrypt...
-- Looking for libgcrypt... /usr/lib/libgcrypt.so
--   found libgcrypt, version 
CMake Error at nasl/CMakeLists.txt:103 (message):
  libgcrypt 1.6 or greater is required


-- Looking for bison...
-- Looking for libgcrypt...
-- Looking for libgcrypt... /usr/lib/libgcrypt.so
--   found libgcrypt, version 
CMake Error at src/CMakeLists.txt:38 (message):
  libgcrypt 1.6 or greater is required


-- Looking for pcap...
-- Looking for pcap... /usr/lib/libpcap.so
-- Looking for pcap-config...
-- Looking for pcap-config... /usr/bin/pcap-config
-- WARNING: pandoc is required to build the HTML user manual.
-- Configuring incomplete, errors occurred!
```

# Environment

**Operating system:**

Arch Linux

`uname -a` shows:

```
Linux ryle 6.9.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 12 Jul 2024 00:06:53 +0000 x86_64 GNU/Linux
```

`cat /etc/lsb-release` shows

```
DISTRIB_ID="Arch"
DISTRIB_RELEASE="rolling"
DISTRIB_DESCRIPTION="Arch Linux"
```

`pacman -Q | \grep libgcrypt` shows

```
lib32-libgcrypt 1.11.0-1
libgcrypt 1.11.0-2
```

`pacman -Ql libgcrypt` shows:

```
libgcrypt /usr/
libgcrypt /usr/bin/
libgcrypt /usr/bin/dumpsexp
libgcrypt /usr/bin/hmac256
libgcrypt /usr/bin/mpicalc
libgcrypt /usr/include/
libgcrypt /usr/include/gcrypt.h
libgcrypt /usr/lib/
libgcrypt /usr/lib/libgcrypt.so
libgcrypt /usr/lib/libgcrypt.so.20
libgcrypt /usr/lib/libgcrypt.so.20.5.0
libgcrypt /usr/lib/pkgconfig/
libgcrypt /usr/lib/pkgconfig/libgcrypt.pc
libgcrypt /usr/share/
libgcrypt /usr/share/aclocal/
libgcrypt /usr/share/aclocal/libgcrypt.m4
libgcrypt /usr/share/info/
libgcrypt /usr/share/info/gcrypt.info-1.gz
libgcrypt /usr/share/info/gcrypt.info-2.gz
libgcrypt /usr/share/info/gcrypt.info.gz
libgcrypt /usr/share/licenses/
libgcrypt /usr/share/licenses/libgcrypt/
libgcrypt /usr/share/licenses/libgcrypt/COPYING.LIB
libgcrypt /usr/share/licenses/libgcrypt/LICENSES
libgcrypt /usr/share/man/
libgcrypt /usr/share/man/man1/
libgcrypt /usr/share/man/man1/hmac256.1.gz
```

<!-- e.g. paste output of uname -a and cat /etc/lsb-release -->

**Installation method / source:** (packages, source installation)

`main` branch as of right now: 1ffc0c83.

# Logfiles

None

--- a/nasl/CMakeLists.txt
+++ b/nasl/CMakeLists.txt
@@ -84,26 +84,7 @@ if (NETSNMP)
 endif (NETSNMP)
 
 message (STATUS "Looking for libgcrypt...")
-find_library (GCRYPT gcrypt)
-message (STATUS "Looking for libgcrypt... ${GCRYPT}")
-if (NOT GCRYPT)
-  message (SEND_ERROR "The libgcrypt library is required.")
-else (NOT GCRYPT)
-  execute_process (COMMAND libgcrypt-config --libs
-    OUTPUT_VARIABLE GCRYPT_LDFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND libgcrypt-config --cflags
-    OUTPUT_VARIABLE GCRYPT_CFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND libgcrypt-config --version
-    OUTPUT_VARIABLE GCRYPT_VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message (STATUS "  found libgcrypt, version ${GCRYPT_VERSION}")
-  if (GCRYPT_VERSION VERSION_LESS "1.6")
-    message (SEND_ERROR "libgcrypt 1.6 or greater is required")
-  endif (GCRYPT_VERSION VERSION_LESS "1.6")
-  string(REPLACE "-I" "" GCRYPT_INCLUDE_DIRS "${GCRYPT_CFLAGS}")
-endif (NOT GCRYPT)
+pkg_check_modules (GCRYPT REQUIRED libgcrypt)
 
 
 ## Library

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -19,26 +19,7 @@ pkg_check_modules (GLIB REQUIRED glib-2.0>=2.42)
 pkg_check_modules (LIBBSD REQUIRED libbsd)
 
 message (STATUS "Looking for libgcrypt...")
-find_library (GCRYPT gcrypt)
-if (NOT GCRYPT)
-  message (SEND_ERROR "The libgcrypt library is required.")
-else (NOT GCRYPT)
-  message (STATUS "Looking for libgcrypt... ${GCRYPT}")
-  execute_process (COMMAND libgcrypt-config --libs
-    OUTPUT_VARIABLE GCRYPT_LDFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND libgcrypt-config --cflags
-    OUTPUT_VARIABLE GCRYPT_CFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND libgcrypt-config --version
-    OUTPUT_VARIABLE GCRYPT_VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message (STATUS "  found libgcrypt, version ${GCRYPT_VERSION}")
-  if (GCRYPT_VERSION VERSION_LESS "1.6")
-    message (SEND_ERROR "libgcrypt 1.6 or greater is required")
-  endif (GCRYPT_VERSION VERSION_LESS "1.6")
-  string(REPLACE "-I" "" GCRYPT_INCLUDE_DIRS "${GCRYPT_CFLAGS}")
-endif (NOT GCRYPT)
+pkg_check_modules (GCRYPT REQUIRED libgcrypt)
 
 message (STATUS "Looking for pcap...")
 find_library (PCAP pcap)

